integration_tests/suite/test_cdr_recording.py

```diff
@@ -284,3 +285,181 @@ def test_delete_media_using_token_tenant_query_string(self):
+            raises(CallLogdError).matching(
+                has_properties(status_code=404, error_id='cdr-not-found-with-given-id')
+            ),
+        )
+
+    @call_log(**{'id': 1}, recordings=[{'path': None}])
+    @call_log(**{'id': 2}, recordings=[{'path': None}])
+    def test_delete_media_with_recording_already_deleted_multi_cdr(self):
+        self.call_logd.cdr.delete_cdrs_recording_media([1, 2])
+        recording = self.call_logd.cdr.get_by_id(1)['recordings'][0]
+        assert_that(recording['deleted'], equal_to(True))
+        recording = self.call_logd.cdr.get_by_id(2)['recordings'][0]
+        assert_that(recording['deleted'], equal_to(True))
+
+    @call_log(**{'id': 1}, recordings=[{'path': '/tmp/denied.wav'}])
+    @call_log(**{'id': 2}, recordings=[{'path': '/tmp/ok.wav'}])
+    def test_delete_media_when_file_has_wrong_permission_multi_cdr(self):
+        self.filesystem.create_file('/tmp/denied.wav', mode='000', root=True)
+        self.filesystem.create_file('/tmp/ok.wav')
+        assert_that(
+            calling(self.call_logd.cdr.delete_cdrs_recording_media).with_args([1, 2]),
+            raises(CallLogdError).matching(
+                has_properties(
+                    status_code=500, error_id='recording-media-permission-denied'
+                )
+            ),
+        )
+
+    @call_log(**{'id': 1}, recordings=[{'path': '/tmp/deleted1.wav'}])
+    @call_log(**{'id': 2}, recordings=[{'path': '/tmp/deleted2.wav'}])
+    def test_delete_media_when_file_deleted_on_filesystem_multi_cdr(self):
+        self.call_logd.cdr.delete_cdrs_recording_media([1, 2])
+        recording = self.call_logd.cdr.get_by_id(1)['recordings'][0]
+        assert_that(recording['deleted'], equal_to(True))
+        recording = self.call_logd.cdr.get_by_id(2)['recordings'][0]
+        assert_that(recording['deleted'], equal_to(True))
+
+    @call_log(
+        **{'id': 10},
+        tenant_uuid=SUB_TENANT,
+        recordings=[{'path': '/tmp/10-recording.wav'}],
+    )
+    @call_log(
+        **{'id': 11},
+        tenant_uuid=MAIN_TENANT,
+        recordings=[{'path': '/tmp/11-recording.wav'}],
+    )
+    def test_delete_media_mutli_tenant_multi_cdr(self):
+        self.filesystem.create_file('/tmp/10-recording.wav', content='10-recording')
+        self.filesystem.create_file('/tmp/11-recording.wav', content='11-recording')
+
+        assert_that(
+            calling(self.call_logd.cdr.delete_cdrs_recording_media).with_args(
+                [10, 11], tenant_uuid=SUB_TENANT
+            ),
+            raises(CallLogdError).matching(
+                has_properties(status_code=404, error_id='cdr-not-found-with-given-id')
+            ),
+        )
+
+        recording = self.call_logd.cdr.get_by_id(10)['recordings'][0]
+        assert_that(recording, has_entries(deleted=False))
+        recording = self.call_logd.cdr.get_by_id(11)['recordings'][0]
+        assert_that(recording, has_entries(deleted=False))
```

Comment:
I did that because I do not want anything to be deleted if one of the provided CDR is not found. My first implementation deleted the recordings from the valid CDRs even though the API raised an exception one one of the CDR.